idevice/src/services/mobilebackup2.rs

```diff
@@ -0,0 +1,1115 @@
+        
+        // Check for error in response
+        if let Some(error) = response.get("ErrorCode") {
+            warn!("Backup start failed with error: {error:?}");
+            return Err(IdeviceError::UnexpectedResponse);
+        }
+        
+        debug!("Backup started successfully");
+        Ok(())
+    }
+
+    /// Starts a restore operation
+    ///
+    /// # Arguments
+    /// * `backup_uuid` - UUID of the backup to restore from
+    /// * `options` - Optional restore options
+    ///
+    /// # Returns
+    /// `Ok(())` on successful restore start
+    ///
+    /// # Errors
+    /// Returns `IdeviceError` if the restore fails to start
+    #[deprecated(note = "Use restore_from_path; restore via BackupUUID is not supported by device/mobilebackup2")]
+    pub async fn start_restore(
+        &mut self,
+        _backup_uuid: &str,
+        options: Option<Dictionary>,
+    ) -> Result<(), IdeviceError> {
+        let mut opts = options.unwrap_or_default();
+        // Align default restore options with pymobiledevice semantics
+        // Caller-specified values (if any) take precedence.
+        if !opts.contains_key("RestoreShouldReboot") {
+            opts.insert("RestoreShouldReboot".into(), plist::Value::Boolean(true));
+        }
+        if !opts.contains_key("RestoreDontCopyBackup") {
+            // pymobiledevice: copy=True -> RestoreDontCopyBackup=False
+            opts.insert("RestoreDontCopyBackup".into(), plist::Value::Boolean(false));
+        }
+        if !opts.contains_key("RestorePreserveSettings") {
+            opts.insert("RestorePreserveSettings".into(), plist::Value::Boolean(true));
+        }
+        if !opts.contains_key("RestoreSystemFiles") {
+            opts.insert("RestoreSystemFiles".into(), plist::Value::Boolean(false));
+        }
+        if !opts.contains_key("RemoveItemsNotRestored") {
+            opts.insert("RemoveItemsNotRestored".into(), plist::Value::Boolean(false));
+        }
+        // Avoid borrowing self while sending request
+        let target_udid_owned = self.idevice.udid().map(|s| s.to_string());
+        let target_udid = target_udid_owned.as_deref();
+        self.send_request(
+            BackupMessageType::BackupMessageTypeRestore.as_str(),
+            // default identifiers to current UDID if available
+            target_udid,
+            target_udid,
+            Some(opts),
+        )
+        .await?;
+        
+        let response = self.receive_backup_response().await?;
+        
+        // Check for error in response
+        if let Some(error) = response.get("ErrorCode") {
+            warn!("Restore start failed with error: {error:?}");
+            return Err(IdeviceError::UnexpectedResponse);
+        }
+        
+        debug!("Restore started successfully");
+        Ok(())
+    }
+
+    /// High-level API: Restore from a local backup directory using DeviceLink file exchange
+    ///
+    /// - `backup_root` 应指向备份根目录（包含 `<SourceIdentifier>` 子目录）
```

Comment:
For consistency and so all contributors can understand and maintain the code, could you translate the comments and function definitions into English? We try to keep all public APIs and documentation in English so the whole community can follow along.